nca/FWRules/ConnectivityGraph.py

```diff
@@ -197,32 +300,60 @@ def get_connectivity_dot_format_str(self, connectivity_restriction=None):
+            # in this case, we do not want to set it as MultiPod, so we check the text size and not group size:
+            node_type = DotGraph.NodeType.MultiPod if len(text) > 1 else node_type
             dot_graph.add_node(nc_name, peer_name, node_type, text)
-
+            # adding the self edges:
+            if len(text) > 1:
+                for connections in peer_connections:
+                    conn_str = connections.get_simplified_connections_representation(True)
+                    conn_str = conn_str.replace("Protocol:", "").replace('All connections', 'All')
+                    dot_graph.add_edge(peer_name, peer_name, label=conn_str, is_dir=False)
+
+        representing_peers = [multi_peer[0][0] for multi_peer in peers_groups]
         for connections, peer_pairs in self.connections_to_peers.items():
             directed_edges = set()
             # todo - is there a better way to get edge details?
             # we should revisit this code after reformatting connections labels
             conn_str = connections.get_simplified_connections_representation(True)
             conn_str = conn_str.replace("Protocol:", "").replace('All connections', 'All')
             for src_peer, dst_peer in peer_pairs:
-                if src_peer != dst_peer and connections:
-                    src_peer_name, _, src_nc = self._get_peer_details(src_peer)
-                    dst_peer_name, _, dst_nc = self._get_peer_details(dst_peer)
+                if src_peer != dst_peer and connections and src_peer in representing_peers and dst_peer in representing_peers:
+                    src_peer_name, _, src_nc, _ = self._get_peer_details(src_peer)
+                    dst_peer_name, _, dst_nc, _ = self._get_peer_details(dst_peer)
                     directed_edges.add(((src_peer_name, src_nc), (dst_peer_name, dst_nc)))
 
-            directed_edges, not_directed_edges, new_peers = self._creates_cliqued_graph(directed_edges)
+            # replacing directed edges with not directed edges:
+            not_directed_edges = set(edge for edge in directed_edges if (edge[1], edge[0]) in directed_edges)
+            directed_edges = directed_edges - not_directed_edges
+            not_directed_edges = set(edge for edge in not_directed_edges if edge[1] < edge[0])
+
+            not_directed_edges, cliques_edges, new_cliques = self._creates_cliqued_graph(not_directed_edges, conn_str)
+            directed_edges = directed_edges | not_directed_edges | set((edge[1], edge[0]) for edge in not_directed_edges)
+
+            directed_edges, bicliques_edges, new_bicliques = self._creates_bicliqued_graph(directed_edges, conn_str)
+
+            # replacing directed edges with not directed edges:
+            not_directed_edges = set(edge for edge in directed_edges if (edge[1], edge[0]) in directed_edges)
+            directed_edges = directed_edges - not_directed_edges
+            not_directed_edges = set(edge for edge in not_directed_edges if edge[1] < edge[0])
```

Comment:
duplicate code. maybe externalize to a separate function.

Reply:
done